torchvision/datasets/omniglot.py

```diff
@@ -0,0 +1,195 @@
+            [
+                [
+                    os.path.join(alphabet, character)
+                    for character in list_dir(os.path.join(self.target_folder, alphabet))
+                ]
+                for alphabet in self._alphabets
+            ],
+            []
+        )
+        self._character_images = [
+            [
+                (image, idx)
+                for image in list_files(os.path.join(self.target_folder, character), '.png')
+            ]
+            for idx, character in enumerate(self._characters)
+        ]
+        self._flat_character_images = sum(self._character_images, [])
+
+    def __len__(self):
+        return len(self._flat_character_images)
+
+    def __getitem__(self, index):
+        """
+        Args:
+            index (int): Index
+
+        Returns:
+            tuple: (image, target) where target is index of the target character class.
+        """
+        image_name, character_class = self._flat_character_images[index]
+        image_path = os.path.join(self.target_folder, self._characters[character_class], image_name)
+        image = Image.open(image_path, mode='r').convert('L')
+
+        if self.transform:
+            image = self.transform(image)
+
+        if self.target_transform:
+            character_class = self.target_transform(character_class)
+
+        return image, character_class
+
+    def _check_integrity(self):
+        zip_filename = self._get_target_folder()
+        if not check_integrity(os.path.join(self.root, zip_filename + '.zip'), self.zips_md5[zip_filename]):
+            return False
+        return True
+
+    def download(self):
+        import zipfile
+
+        if self._check_integrity():
+            print('Files already downloaded and verified')
+            return
+
+        filename = self._get_target_folder()
+        zip_filename = filename + '.zip'
+        url = self.download_url_prefix + '/' + zip_filename
+        download_url(url, self.root, zip_filename, self.zips_md5[filename])
+        print('Extracting downloaded file: ' + os.path.join(self.root, zip_filename))
+        with zipfile.ZipFile(os.path.join(self.root, zip_filename), 'r') as zip_file:
+            zip_file.extractall(self.root)
+
+    def _get_target_folder(self):
+        return 'images_background' if self.background is True else 'images_evaluation'
+
+
+class OmniglotRandomPair(Omniglot):
+    """`OmniglotRandomPair <https://github.com/brendenlake/omniglot>`_ Dataset.
+
+    This is a subclass of the Omniglot dataset. This instead it returns
+    a randomized pair of images with similarity label (0 or 1)
+
+    Args:
+        pair_count (int, optional): The total number of image pairs to generate. Defaults to
+            10000
+    """
+    def __init__(self, root, pair_count=10000, background=True,
+                 transform=None, target_transform=None,
+                 download=False):
+        super(self.__class__, self).__init__(root, background=background,
```

Comment:
usually we dont use `self.__class__` but instead do `super(OmniglotRandomPair, self).__init__(...)`

Reply:
Fair enough. Though any major reason why `self.__class__` would be a bad practice?